pandas/core/series.py

```diff
@@ -771,16 +778,20 @@ def take(self, indices, axis=0, is_copy=False, **kwargs) -> "Series":
             kwargs = {}
         new_values = self._values.take(indices, **kwargs)
 
-        result = self._constructor(
+        return self._constructor(
             new_values, index=new_index, fastpath=True
         ).__finalize__(self)
 
-        # Maybe set copy if we didn't actually change the index.
-        if is_copy:
-            if not result._get_axis(axis).equals(self._get_axis(axis)):
-                result._set_is_copy(self)
+    def _take_with_is_copy(self, indices, axis=0, **kwargs):
+        """
+        Internal version of the `take` method that sets the `_is_copy`
+        attribute to keep track of the parent dataframe (using in indexing
+        for the SettingWithCopyWarning). For Series this does the same
+        as the public take (it never sets `_is_copy`).
 
-        return result
+        See the docstring of `take` for full explanation of the parameters.
+        """
+        return self.take(indices=indices, axis=axis, **kwargs)
```

Comment:
I am including a `Series._take_with_is_copy` to override the generic one. This is because the defaults for the old `is_copy` kwargs were different for Series (False) and DataFrame (True).

I am not fully sure if the code paths that use `_take_with_is_copy` actually can have a Series calling it, but included this to be sure.